app/celery/process_letter_client_response_tasks.py

```diff
@@ -0,0 +1,56 @@
+from flask import current_app
+
+from app import notify_celery
+from app.constants import DVLA_NOTIFICATION_DISPATCHED, NOTIFICATION_DELIVERED, NOTIFICATION_TECHNICAL_FAILURE
+from app.dao.notifications_dao import (
+    dao_get_notification_or_history_by_id,
+    dao_update_notification,
+)
+from app.exceptions import NotificationTechnicalFailureException
+
+
+@notify_celery.task(bind=True, name="process-letter-callback")
+def process_letter_callback_data(self, notification_id, page_count, dvla_status):
```

Comment:
is there a requirement to ensure dvla_status is one of DISPATCHED and REJECTED before processing?

Reply:
yes, we do this as part of the schema validation here: https://github.com/alphagov/notifications-api/blob/91ee6fa3410d912f1c91c05465bd3952d9244bc6/app/notifications/notifications_letter_callback.py#L88